jsonpickle/util.py

```diff
@@ -243,21 +243,14 @@ def is_function(obj):
     >>> is_function(1)
     False
     """
-    if type(obj) in (types.FunctionType,
-                     types.MethodType,
-                     types.LambdaType,
-                     types.BuiltinFunctionType,
-                     types.BuiltinMethodType):
-        return True
-    if not hasattr(obj, '__class__'):
-        return False
-    module = translate_module_name(obj.__class__.__module__)
-    name = obj.__class__.__name__
-    return (module == '__builtin__' and
-            name in ('function',
-                     'builtin_function_or_method',
-                     'instancemethod',
-                     'method-wrapper'))
+    function_types = (
+        types.FunctionType,
+        types.MethodType,
+        types.LambdaType,
+        types.BuiltinFunctionType,
+        types.BuiltinMethodType,
+    )
+    return type(obj) in function_types
```

Comment:
This seems like it should do the right thing.  Would it be simpler still as `isinstance(obj, function_types)`?  Merged as-is nonetheless.  If we discover anything that relies on the original implementation we'll add a test then.  I suspect these weren't available in python 2.5 or 2.6 back when it was written.